### PR TITLE
Added a script to autotranslate en to it,fr,de,es.

### PR DESCRIPTION
**Pull request by kappu72**
This pull add's a script that can automatically translate any missing translations in it-fr-es-de languages.
It checks if the messages in data.en-US.json are missing in other mandatories languages and translate them with google translate service. The script needs a valid google translation apikey
See https://cloud.google.com/translate/docs/basic/setup-basic to setup a project and create apikey credential.
This is relate to the action discussed in our last meeting. Let me know what you think about it.
## Description

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [X] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [X] Other... Please describe:
Developer tool.
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,6 +84,7 @@
     "@geosolutions/proj4": "2.4.7",
     "@geosolutions/react-joyride": "1.10.2",
     "@geosolutions/wkt-parser": "1.2.2",
+    "@google-cloud/translate": "^5.3.0",
```
Comment (MV88) [inline package.json:87]:
```suggestion
    "@google-cloud/translate": "5.3.0",
```

File: utility/translations/autoTranslate.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+const TRANSLATIONS_FOLDER = 'web/client/translations/';
+
+const LANGUAGES = [{code: "it", file: "data.it-IT.json", missing: []}, { code: "es", file: "data.es-ES.json", missing: []}, { code: "de", file: "data.de-DE.json",  missing: []}, { code: "fr", file: "data.fr-FR.json",  missing: []}];
+const log = s => process.stdout.write(`${s}\n`);
+const apykey = process.argv[2];
+
+
+// const fs = require('fs');
+// const baseobj = JSON.parse(fs.readFileSync(`${TRANSLATIONS_FOLDER}${baseFile}`, 'utf8'));
```
Comment (MV88) [inline utility/translations/autoTranslate.js:33]:
```suggestion
```

File: utility/translations/autoTranslate.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    // Group missing translations by language
+    for (const key of flattenObject(baseLanguage)) {
+        missingLaguanges(key);
+    }
+    // Translate single
+    for (const lang of LANGUAGES) {
+        translateLanguage(lang, baseLanguage).then(updated => {
+            if (updated) {
+                saveFile("./" + TRANSLATIONS_FOLDER  + lang.file, lang.data).then(() => log(`Aggiornata e Salvata lingua ${lang.code}`));
```
Comment (MV88) [inline utility/translations/autoTranslate.js:100]:
```suggestion
                saveFile("./" + TRANSLATIONS_FOLDER  + lang.file, lang.data).then(() => log(`Updated and saved locale ${lang.code}`));
```

File: package.json (hunk 1 above)
Comment (kappu72) [reply package.json:87]: Also moved in devDependencies

**Comment by offtherailz** [comment]
I'm going to close this issue, we will provide with as an external script/lib/tool

**Pull request closed without merging**